internal/service/loadbalancer/frontend_tls_config.go

```diff
@@ -2,153 +2,261 @@ package loadbalancer
-		return diag.FromErr(err)
+func (r *frontendTLSConfigResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data frontendTLSConfigModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	t, err := svc.GetLoadBalancerFrontendTLSConfig(ctx, &request.GetLoadBalancerFrontendTLSConfigRequest{
-		ServiceUUID:  serviceID,
-		FrontendName: feName,
-		Name:         name,
-	})
-	if err != nil {
-		return utils.HandleResourceError(d.Get("name").(string), d, err)
+
+	if data.ID.ValueString() == "" {
+		resp.State.RemoveResource(ctx)
+
+		return
 	}
 
-	if err = d.Set("frontend", utils.MarshalID(serviceID, feName)); err != nil {
-		return diag.FromErr(err)
+	var loadbalancer, frontendName string
+	err := utils.UnmarshalID(data.Frontend.ValueString(), &loadbalancer, &frontendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer frontend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
 
-	if diags = setFrontendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.GetLoadBalancerFrontendTLSConfig(ctx, &request.GetLoadBalancerFrontendTLSConfigRequest{
+		ServiceUUID:  loadbalancer,
+		FrontendName: frontendName,
+		Name:         data.Name.ValueString(),
+	})
+	if err != nil {
+		if utils.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to read loadbalancer frontend TLS config details",
+				utils.ErrorDiagnosticDetail(err),
+			)
+		}
+		return
 	}
 
-	return diags
+	resp.Diagnostics.Append(setFrontendTLSConfigValues(ctx, &data, tlsConfig)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
-func resourceFrontendTLSConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-	var serviceID, feName, name string
-	if err := utils.UnmarshalID(d.Id(), &serviceID, &feName, &name); err != nil {
-		return diag.FromErr(err)
+func (r *frontendTLSConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data frontendTLSConfigModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	var loadbalancer, frontendName string
+	err := utils.UnmarshalID(data.Frontend.ValueString(), &loadbalancer, &frontendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer frontend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
-	t, err := svc.ModifyLoadBalancerFrontendTLSConfig(ctx, &request.ModifyLoadBalancerFrontendTLSConfigRequest{
-		ServiceUUID:  serviceID,
-		FrontendName: feName,
-		Name:         name,
+
+	apiReq := &request.ModifyLoadBalancerFrontendTLSConfigRequest{
+		ServiceUUID:  loadbalancer,
+		FrontendName: frontendName,
+		Name:         data.Name.ValueString(),
```

Comment:
_(Same comment as in backend update)_ Should this name come from the ID (or state) as this should be the current name instead of the planned name?

Reply:
Nice catch, name is now parsed from ID in all occasions for clarity. :pray: